Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/HttpUtility.cs

```diff
@@ -0,0 +1,670 @@
+                throw new ArgumentNullException ("output");
+            }
+            HttpEncoder.Current.HtmlAttributeEncode (s, output);
+        }
+
+        public static string HtmlAttributeEncode (string s)
+        {
+            if (s == null)
+                return null;
+
+            using (var sw = new StringWriter ()) {
+                HttpEncoder.Current.HtmlAttributeEncode (s, sw);
+                return sw.ToString ();
+            }
+        }
+
+        public static string UrlDecode (string str)
+        {
+            return UrlDecode(str, Encoding.UTF8);
+        }
+
+        static char [] GetChars (MemoryStream b, Encoding e)
+        {
+            return e.GetChars (b.GetBuffer (), 0, (int) b.Length);
+        }
+
+        static void WriteCharBytes (IList buf, char ch, Encoding e)
+        {
+            if (ch > 255) {
+                foreach (byte b in e.GetBytes (new char[] { ch }))
+                    buf.Add (b);
+            } else
+                buf.Add ((byte)ch);
+        }
+
+        public static string UrlDecode (string str, Encoding e)
+        {
+            if (null == str)
+                return null;
+
+            if (str.IndexOf ('%') == -1 && str.IndexOf ('+') == -1)
+                return str;
+
+            if (e == null)
+                e = Encoding.UTF8;
+
+            long len = str.Length;
+            var bytes = new List <byte> ();
+            int xchar;
+            char ch;
+
+            for (int i = 0; i < len; i++) {
+                ch = str [i];
+                if (ch == '%' && i + 2 < len && str [i + 1] != '%') {
+                    if (str [i + 1] == 'u' && i + 5 < len) {
+                        // unicode hex sequence
+                        xchar = GetChar (str, i + 2, 4);
+                        if (xchar != -1) {
+                            WriteCharBytes (bytes, (char)xchar, e);
+                            i += 5;
+                        } else
+                            WriteCharBytes (bytes, '%', e);
+                    } else if ((xchar = GetChar (str, i + 1, 2)) != -1) {
+                        WriteCharBytes (bytes, (char)xchar, e);
+                        i += 2;
+                    } else {
+                        WriteCharBytes (bytes, '%', e);
+                    }
+                    continue;
+                }
+
+                if (ch == '+')
+                    WriteCharBytes (bytes, ' ', e);
+                else
+                    WriteCharBytes (bytes, ch, e);
+            }
+
+            byte[] buf = bytes.ToArray ();
+            bytes = null;
+            return e.GetString (buf);
+
+        }
+
+        public static string UrlDecode (byte [] bytes, Encoding e)
+        {
+            if (bytes == null)
+                return null;
+
+            return UrlDecode (bytes, 0, bytes.Length, e);
+        }
+
+        static int GetInt (byte b)
+        {
+            char c = (char) b;
+            if (c >= '0' && c <= '9')
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)